src/components/drawer/Drawer.tsx

```diff
@@ -169,107 +254,216 @@ const Drawer = ({
+      springToValue(translationAnim, position, animationStyle, () =>
+        setIsBackgroundVisible(false)
+      )
+      if (isFullscreen) {
+        springToValue(borderRadiusAnim, BORDER_RADIUS, animationStyle)
+      }
+      if (shouldBackgroundDim) {
+        springToValue(shadowAnim, 0, animationStyle)
+        springToValue(backgroundOpacityAnim, 0, animationStyle)
+      }
+    },
+    [
+      translationAnim,
+      shadowAnim,
+      backgroundOpacityAnim,
+      borderRadiusAnim,
+      isFullscreen,
+      shouldBackgroundDim,
+      animationStyle
+    ]
+  )
 
   useEffect(() => {
     if (isOpen) {
-      slideIn()
+      slideIn(openPosition)
       setIsBackgroundVisible(true)
     } else {
-      slideOut()
+      if (isOpenToInitialOffset) {
+        borderRadiusAnim.setValue(0)
+        slideOut(initialOffsetOpenPosition)
+      } else {
+        slideOut(initialPosition)
+      }
     }
-  }, [slideIn, slideOut, isOpen])
+  }, [
+    slideIn,
+    slideOut,
+    isOpen,
+    openPosition,
+    initialPosition,
+    isOpenToInitialOffset,
+    initialOffsetOpenPosition,
+    borderRadiusAnim
+  ])
 
   const panResponder = PanResponder.create({
     onMoveShouldSetPanResponder: (e, gestureState) => {
       return Math.abs(gestureState.dy) > ON_MOVE_RESPONDER_DY
     },
+    /**
+     * Callback when the user is dragging on the drawer.
+     * There are two primary modes of operation:
+     *  - Normal drawers that open by some UI action and can be swiped away by a user
+     *  - Drawers that are partially showing on the screen (e.g. Now Playing)
+     *    that start at an initial offset and can be dragged open
+     */
     onPanResponderMove: (e, gestureState) => {
-      if (isOpen) {
+      if (isOpen || isOpenToInitialOffset) {
         if (gestureState.dy > 0) {
           // Dragging downwards
-          const newTranslation = openPosition + gestureState.dy
-          attachToDy(translationAnim, newTranslation)(e)
-
           const percentOpen = (drawerHeight - gestureState.dy) / drawerHeight
 
-          const newOpacity = BACKGROUND_OPACITY * percentOpen
-          attachToDy(backgroundOpacityAnim, newOpacity)(e)
+          if (isOpen) {
+            const newTranslation = openPosition + gestureState.dy
+            attachToDy(translationAnim, newTranslation)(e)
 
-          if (isFullscreen) {
-            const newBorderRadius = BORDER_RADIUS * (1 - percentOpen)
+            if (shouldBackgroundDim) {
+              const newOpacity = BACKGROUND_OPACITY * percentOpen
+              attachToDy(backgroundOpacityAnim, newOpacity)(e)
+            }
 
-            attachToDy(borderRadiusAnim, newBorderRadius)(e)
+            if (isFullscreen) {
+              const newBorderRadius = BORDER_RADIUS * (1 - percentOpen)
+              attachToDy(borderRadiusAnim, newBorderRadius)(e)
+            }
+
+            // If we are "closing" the drawer to a offset position
+            if (initialOffsetOpenPosition) {
```

Comment:
Should this be `initialOffsetPosition`? Because `initialOffsetOpenPosition` will always be a number based on the height